src/components/article-body/body-components/raw-html.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import sanitizeHtml from 'sanitize-html';
+
+// To allow additional embeds/iframes, update these arrays based on inclusion tag
+// (iframe vs script)
+const ALLOWED_IFRAME_HOSTNAMES = ['open.spotify.com'];
+const ALLOWED_SCRIPT_SRCS = [
+    'https://platform.twitter.com/widgets.js',
+    'https://www.tiktok.com/embed.js',
+];
+
+const isInlineElement = html => html === '<br />';
+
+const RawHTML = ({ value }) => {
+    const [hasRenderedScript, setHasRenderedScript] = useState(false);
+    const containsScript = value.match(/<script/);
+    const contentRef = useRef();
+    const sanitizedHTML = useMemo(
+        () =>
+            sanitizeHtml(value, {
+                allowedTags: sanitizeHtml.defaults.allowedTags.concat([
+                    'img',
+                    'iframe',
+                    'script',
+                ]),
+                // Need to lock script srcs to include tiktok, spotify, twitter
+                allowedAttributes: false,
+                allowedIframeHostnames: ALLOWED_IFRAME_HOSTNAMES,
+                allowVulnerableTags: true,
+                exclusiveFilter: frame => {
+                    return (
+                        frame.tag === 'script' &&
+                        ALLOWED_SCRIPT_SRCS.includes(frame.src)
+                    );
+                },
```

Comment:
If I'm reading [the documentation ](https://www.npmjs.com/package/sanitize-html#user-content-filters)correctly, won't this exclude all scripts in the `ALLOWED_SCRIPT_SRCS` list? I would imagine we would want to allow only those, not exclude only those right?

Reply:
I'm yet to see an example of this @mcolella14. This is the logic in the our old devhub, I haven't come across the article where scripts are embedded. Will get back to you once I come across one such example.